homeassistant/components/renson_endura_delta/sensorvalue.py

```diff
@@ -0,0 +1,51 @@
+"""Sensor class of the Renson ventilation system."""
+from rensonVentilationLib.fieldEnum import FieldEnum
+from rensonVentilationLib.generalEnum import DataType
+import rensonVentilationLib.renson as renson
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+
+
+class SensorValue(SensorEntity):
+    """Get a sensor data from the Renson API and store it in the state of the class."""
+
+    def __init__(
+        self,
+        description: SensorEntityDescription,
+        field: FieldEnum,
+        renson: renson.RensonVentilation,
+        rawFormat: bool,
+    ):
+        """Initialize class."""
+        super().__init__()
+
+        self._state = None
+        self.entity_description = description
+        self.field = field
+        self.dataType = field.field_type
+        self.renson = renson
+        self.rawFormat = rawFormat
+
+    @property
+    def state(self):
+        """Lookup the state of the sensor and save it."""
+        return self._state
+
+    def update(self):
+        """Save state of sensor."""
+        if self.rawFormat:
+            self._state = self.renson.get_data_string(self.field)
```

Comment:
Is each sensor doing it's own request to the device? Is it possible to get all data at once ?

Reply:
The library that I have written is caching the data from the device for 1 minute. It will only call once every minute to the device itself. HA is just calling the library to translate the json of the device to the real data.